src/renderer/containers/Header/NameChangeDialog.tsx

```diff
@@ -64,15 +71,24 @@ export const NameChangeDialog: React.FC<{
           )
         }
       >
-        <TextField
-          required={true}
-          autoFocus={true}
-          value={name}
-          label="Display name"
-          inputProps={{
-            maxLength: 15,
-          }}
-          onChange={(event) => setName(event.target.value)}
+        <Controller
+          name="displayName"
+          control={control}
+          defaultValue=""
+          render={({ field, fieldState: { error } }) => (
+            <TextField
+              {...field}
+              label="Display Name"
+              required={true}
+              error={Boolean(error)}
+              helperText={error ? error.message : undefined}
+              autoFocus={true}
+              inputProps={{
+                maxLength: 15,
+              }}
+            />
+          )}
+          rules={{ validate: isValidDisplayName }}
```

Comment:
```suggestion
          rules={{ validate: (val) => isValidDisplayName(val.trim()) }}
```

I think we should pass the trimmed version here instead of delegating that to the validation logic.